software/contrib/logic.py

```diff
@@ -44,16 +43,26 @@ def main(self):
         @b1.handler
         def on_b1_press():
             ssoled.notify_user_interaction()
+            self.b1_high = 1
```

Comment:
[optional] Alternatively consider using the Button class's [Button.value()](https://allen-synthesis.github.io/EuroPi/generated/europi.html#europi.Button.value) method instead of setting an instance variable. This will allow you to remove quite a bit of the code in this change.

Reply:
Right, I somehow forgot that `.value()` existed & returned an integer.